src/Chapter2.hs

```diff
@@ -740,7 +764,8 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains :: Eq a => a -> [[a]] -> [[a]]
```

Comment:
At first, I declared the function type as simply a -> [[a]] -> [[a]] then I got a compilation error and an advice from ghc to limit a to typeclass Eq. As far as I understand, it happened due to not every type allows its instances to be compared to each other. In other words, not every type supports equality testing. My function does compare values so their type must be of basic typeclass Eq.

Reply:
Yes, you understand everything absolutely correctly! We will cover this topic in the following chapter, but `==` operation that allows answering about elements equality is the method of the type class. None of the data types has it by default, so if you are using this method, you need to tell the compiler, that here you will do `==` which means that this function only can work with data types that you can compare. And compiler knows that 🙂